CorsixTH/Lua/entities/machine.lua

```diff
@@ -155,8 +177,13 @@ function Machine:machineUsed(room)
     if taskIndex == -1 then
       local call = self.world.dispatcher:callForRepair(self, true, false, true)
       self.hospital:addHandymanTask(self, "repairing", 2, self.tile_x, self.tile_y, call)
+      self:announceRepair(room)
     else -- Otherwise the task is already queued. Increase the priority to above that of machines with at least 4 uses left
-      self.hospital:modifyHandymanTaskPriority(taskIndex, 2, "repairing")
+      local subTable = self.hospital:findHandymanTaskSubtable("repairing")
+      if subTable[taskIndex].priority == 1 then
+        self.hospital:modifyHandymanTaskPriority(taskIndex, 2, "repairing")
```

Comment:
The machine shouldn't hack the internals of the handyman task administration of the hospital.
Instead, wrap this in a new handymanTask function. In other words, be nice, *ask* the hospital to change the priority for you.

The goal is to keep the amount of code that changes some data structure as local and small as possible, with a well-defined (function) interface. In that way, you can change such a part, or move such code by modifying one or two files, instead of 100 files.

Reply:
Somewhat the same is your earthquake queries above.
That is however more of an edge case. I am not sure what part of the data is public and what is internal.

Reply:
I have a feeling having just these 2 priorities for repair tasks is a little lacking when the plants have 5. I'm not expecting to address this now, but maybe something to consider later